### DISCUSSION TITLE
How to configure a webhook with an url that contains a "!" like matrix room has ?

### DISCUSSION THREAD

**Discussion by dtrckd** [postalserver/postal#3241] [Help with using Postal] [answered]
How to configure a webhook with an url that contains a "!" like matrix room has ?

I am currently getting **URL is invalid** on the webui. (postal version 3.3.4)

**Comment by willpower232** [comment, marked as answer]
You would have to modify this line I expect https://github.com/postalserver/postal/blob/fd3c7ccdf6dc4ee0a76c9523cbd735159e4b8000/app/models/webhook.rb#L33

**Reply by dtrckd** [reply]
Hi, thanks @willpower232 

Just to be sure, I have read that postal won't accept new PR; 
- Is it correct ? 
- is this (simple) change in postal might be considered in future release ?

**Reply by willpower232** [reply]
I don't know where you've read that but PRs should always be welcome.

It might be that they don't get a chance to review and ship new features often but being open source, you should be able to create your own docker images with any code changes you require.

**Reply by dtrckd** [reply]
Ok. 

About the idea of creating a new docker image, don't you think it is simple to directly patch the file inside the container instead of creating a new one. It feels easier to apply the change over upgrades ?

**Reply by willpower232** [reply]
Unfortunately the way ruby on rails works means you have to restart the docker container to apply changes and restarting the docker container will remove any change you make back to the original values so you have to make your own docker image to properly test changes.

**Reply by dtrckd** [reply]
I am not sure about your statement. 

I have just tested to make the change in file directly inside the container `postal-smtp-1`, file `/opt/postal/app/app/models/webhook.rb`, then restart postal (it did not recreate the container). 

It seems to work as expected.

**Reply by dtrckd** [reply]
Ok, the matrix API, need a PUT request, not a POST. I guess another modification to postal to support that 😅

**Reply by willpower232** [reply]
Well the change you made is still temporary so could go away at any moment so bear that in mind if you need to edit or add any more.

You would need to modify the worker and web docker containers, I don't think the smtp one is involved.
